programs/jit-proxy/src/lib.rs

```diff
@@ -237,6 +299,73 @@ pub enum PriceType {
     Oracle,
 }
 
+#[derive(Accounts)]
+pub struct CheckOrderConstraints<'info> {
+    pub user: AccountLoader<'info, User>,
+}
+
+#[derive(Debug, Clone, Copy, AnchorSerialize, AnchorDeserialize, PartialEq, Eq)]
+pub struct OrderConstraint {
+    pub max_position: i64,
+    pub min_position: i64,
+    pub market_index: u16,
+    pub market_type: MarketType,
+}
+
+impl OrderConstraint {
+    pub fn check(&self, current_position: i64, open_bids: i64, open_asks: i64) -> Result<()> {
+        let max_long = current_position.safe_add(open_bids)?;
+
+        if max_long > self.max_position {
+            msg!(
+                "market index {} market type {:?}",
+                self.market_index,
+                self.market_type
+            );
+            msg!(
+                "max long {} current position {} open bids {}",
+                max_long,
+                current_position,
+                open_bids
+            );
+            return Err(ErrorCode::OrderSizeBreached.into());
+        }
+
+        let max_short = current_position.safe_add(open_asks)?;
+        if max_short < self.min_position {
+            msg!(
+                "market index {} market type {:?}",
+                self.market_index,
+                self.market_type
+            );
+            msg!(
+                "max short {} current position {} open asks {}",
+                max_short,
+                current_position,
+                open_asks
+            );
+            return Err(ErrorCode::OrderSizeBreached.into());
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Copy, BorshSerialize, BorshDeserialize, PartialEq, Debug, Eq)]
+pub enum MarketType {
+    Perp,
+    Spot,
+}
```

Comment:
why did you have to do recreate this rather than import?

Reply:
annoying anchor thing, to get MarketType in jit idl, needs to be declared here